lib/gateway.js

```diff
@@ -146,6 +144,25 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
   }
 }
 
+function makeEntityTransformData (field) {
+  const isNonNull = field.astNode.type.kind === Kind.NON_NULL_TYPE
+  const leafKind = isNonNull ? field.astNode.type.type.kind : field.astNode.type.kind
+
+  if (leafKind === Kind.LIST_TYPE) {
+    return function transformData (response) {
+      if (!response) {
+        return isNonNull ? [] : null
+      }
+
+      return response.json.data._entities
+    }
```

Comment:
I would prefer if we did not create a closure over the ast node, can you move this to two top level functions?